greenheart/simulation/technologies/hydrogen/h2_transport/h2_export_pipe.py

```diff
@@ -2,167 +2,296 @@
-    if S_h>=S_h_check:
+    rho_water = 1_000  # kg/m3
+    p_hydrostatic = rho_water * 9.81 * depth /BAR2PA  # bar
+    dP = (p_inlet - p_hydrostatic) * BAR2MPA  # MPa
+    S_h = (
+        dP * (diam - (thickness if diam / thickness >= 30 else 0)) / (2_000 * thickness)
+    )
+    if S_h >= S_h_check:
         return False
 
     #   Longitudinal stress (MPa)
-    S_L_check = 0.8*SMYS #2020 ASME B31.8 Table A842.2.2-1. Same for riser and pipe
-    S_L = p_inlet*bar2MPa*(diam-2*thickness)/(4*thickness)
-    if S_L>S_L_check:
+    S_L_check = 0.8 * SMYS  # 2020 ASME B31.8 Table A842.2.2-1. Same for riser and pipe
+    S_L = p_inlet * BAR2MPA * (diam - 2 * thickness) / (4 * thickness)
+    if S_L > S_L_check:
         return False
 
-    S_combined_check = 0.9*SMYS #2020 ASME B31.8 Table A842.2.2-1. Same for riser and pipe
+    S_combined_check = (
+        0.9 * SMYS
+    )  # 2020 ASME B31.8 Table A842.2.2-1. Same for riser and pipe
     #   Torsional stress?? Under what applied torque? Not sure what to do for this.
 
     return True
 
-def get_anl_costs(costs,total_L):
-    labor_coef = [95295,0.53848,0.03070]
-    misc_coef = [19211,0.14178,0.04697]
-    row_coef = [72634,1.07566,0.05284]
-    mat_coef = [5605,0.41642,0.06441] # Don't need this anymore since material cost is coming from Savoy numbers
-
-    L_mi = total_L*0.621371
 
-    # costs['mat cost anl [$]'] = costs.apply(lambda x: (mat_coef[0]*((x['Outer diameter (mm)']*mm2in)**mat_coef[1])/L_mi**mat_coef[2])*(x['Outer diameter (mm)']*mm2in*L_mi),axis=1)
-    costs['labor cost [$]'] = costs.apply(lambda x: (labor_coef[0]/((x['Outer diameter (mm)']*mm2in)**labor_coef[1])*L_mi**labor_coef[2])*(x['Outer diameter (mm)']*mm2in*L_mi),axis=1)
-    costs['misc cost [$]'] = costs.apply(lambda x: (misc_coef[0]/((x['Outer diameter (mm)']*mm2in)**misc_coef[1])/L_mi**misc_coef[2])*(x['Outer diameter (mm)']*mm2in*L_mi),axis=1)
-    costs['ROW cost [$]'] = costs.apply(lambda x: (row_coef[0]/((x['Outer diameter (mm)']*mm2in)**row_coef[1])*L_mi**row_coef[2])*(x['Outer diameter (mm)']*mm2in*L_mi),axis=1)
+def get_anl_costs(costs: pd.DataFrame, total_L: float) -> pd.DataFrame:
+    labor_coef = [95295, 0.53848, 0.03070]
+    misc_coef = [19211, 0.14178, 0.04697]
+    row_coef = [72634, 1.07566, 0.05284]
+    mat_coef = [
+        5605,
+        0.41642,
+        0.06441,
+    ]  # Don't need this anymore since material cost is coming from Savoy numbers
 
-    costs['total capital cost [$]'] = costs[['mat cost [$]','labor cost [$]','misc cost [$]','ROW cost [$]']].sum(axis=1)
+    L_mi = total_L * 0.621371
 
-    costs['annual operating cost [$]'] = 0.0117*costs['total capital cost [$]'] # https://doi.org/10.1016/j.esr.2021.100658
+    # costs['mat cost anl [$]'] = costs.apply(lambda x: (mat_coef[0]*((x['Outer diameter (mm)']*mm2in)**mat_coef[1])/L_mi**mat_coef[2])*(x['Outer diameter (mm)']*mm2in*L_mi),axis=1)
+    costs["labor cost [$]"] = costs.apply(
```

Comment:
Could you add a few more comments so readers can understand the costs calculations more readily?